raft: nil pointer dereference in raft.stepLeader

The following test appears to have failed:

[#8754](https://circleci.com/gh/cockroachdb/cockroach/8754):

```
I1029 20:15:45.963365 955 raft/raft.go:409  [group 1] 8 became follower at term 1
I1029 20:15:45.966295 955 raft/raft.go:503  [group 1] 8 [term: 1] received a MsgApp message with higher term from 1 [term: 6]
I1029 20:15:45.966408 955 raft/raft.go:409  [group 1] 8 became follower at term 6
W1029 20:15:45.966729 955 multiraft/multiraft.go:1041  failed to look up replica ID for range 1 (disabling replica ID check): storage/store.go:1569: store 3 not found as replica of range 1
I1029 20:15:45.973107 955 storage/store.go:1472  got an event membership change committed event: EndTransaction ["\x00\x00\x00k1\x00\x01rdsc","")
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x10 pc=0xdcd650]

goroutine 6862 [running]:
github.com/coreos/etcd/raft.stepLeader(0xc820462ea0, 0xb, 0x0, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/coreos/etcd/raft/raft.go:590 +0x20e0
github.com/coreos/etcd/raft.(*raft).Step(0xc820462ea0, 0xb, 0x0, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/coreos/etcd/raft/raft.go:511 +0x380
github.com/coreos/etcd/raft.(*multiNode).run(0xc8205d6a80)
	/go/src/github.com/coreos/etcd/raft/multinode.go:258 +0x1667
created by github.com/coreos/etcd/raft.StartMultiNode
--
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:595 +0x447
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820451aa0, 0xc8203abfc0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
FAIL	github.com/cockroachdb/cockroach/storage	19.358s
=== RUN   TestBatchBasics
I1029 20:15:24.220419 949 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchGet
I1029 20:15:24.224933 949 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I1029 20:15:24.231714 949 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/GLOCKFILE
+++ b/GLOCKFILE
@@ -19,7 +19,7 @@ github.com/cockroachdb/c-protobuf 6a18bfcdd5169966cd1235edb71d749c950216e8
 github.com/cockroachdb/c-rocksdb bf15ead80bdc205a19b3d33415b23c156a3cf371
 github.com/cockroachdb/c-snappy 5c6d0932e0adaffce4bfca7bdf2ac37f79952ccf
 github.com/cockroachdb/yacc 443154b1852a8702b07d675da6cd97cd9177a316
-github.com/coreos/etcd a423a55b142c2b9a82811604204cddbccd0a9cf9
+github.com/coreos/etcd 977d3615f79d420c839f9b10ec994c82bc18a00d
 github.com/cpuguy83/go-md2man 71acacd42f85e5e82f70a55327789582a5200a90
 github.com/docker/docker 758fae5f6b079b0125a657854e7e853d638dc569
 github.com/elazarl/go-bindata-assetfs d5cac425555ca5cf00694df246e04f05e6a55150

